components/markdown/src/markdown.rs

```diff
@@ -793,15 +796,50 @@ pub fn markdown_to_html(
             .position(|e| matches!(e, Event::Html(CowStr::Borrowed(CONTINUE_READING))))
             .unwrap_or(events.len());
 
+        // determine closing tags missing from summary
```

Comment:
Could we error if the summary marker is _inside_ a tag? Eg `<i <-- more --> class='' />`

Reply:
Since that's just straight-up invalid HTML, I'd be more inclined to just make the `<!-- more -->` regexp only match if it's by itself (i.e. not nested inside something like that). Will add a test for it.

Personally, I think that an error would be weird; we don't emit errors for invalid HTML otherwise, so, it's better to just ignore it and treat it like otherwise malformed HTML.

Reply:
Fair enough

Reply:
So, I tried adding a test for this, but the markdown parser just isn't happy with it. If it's valid HTML, it parses, and if it's not, it just escapes it. So, in your example, it shows up as `&lt;i` at the end of the summary.

I'm inclined to just say that's okay, honestly. We could detect if it's inside any custom HTML tags, like `<i><!-- MORE --></i>`, but that goes against the principle that we aren't accounting for custom HTML tags at all and those just remain opened.

Reply:
Sounds good